gen/common/atom.mli

```diff
@@ -25,14 +26,39 @@ module type SIMD = sig
     val reduce: int list list -> int
 end
 
-module type S = sig
-  val bellatom : bool (* true if bell style atoms *)
-
+module type RMW = sig
+  (* The `rmw` edge *)
+  type rmw
+  (* Types `atom` and `value` should be passed from outside *)
   type atom
 
-(* SIMD writes and reads *)
-  module SIMD : SIMD
+  val pp_rmw : bool (* backward compatibility *) -> rmw -> string
+  val is_one_instruction : rmw -> bool
+  (* The first boolean indicates whether wildcard syntax is included in the fold *)
+  val fold_rmw : bool -> (rmw -> 'a -> 'a) -> 'a -> 'a
+  (* Second round of fold, for rmw with back compatible name *)
+  val fold_rmw_compat : (rmw -> 'a -> 'a) -> 'a -> 'a
+  val applies_atom_rmw : rmw -> atom option -> atom option -> bool
+  val show_rmw_reg : rmw -> bool
+  val compute_rmw : rmw  -> int (* old *) -> int (* operand *) -> int
```

Comment:
In situations like this, where a function takes multiple arguments of the same type and their meaning is not clear from context, it is better to disambiguate them using names arguments (which are enforced by the compiler) rather than comments (which aren't enforced and may drift away from the actual implementation with time).
 
```suggestion
  val compute_rmw : rmw  -> old:int -> operand:int -> int
```